distributed/tests/test_client.py

```diff
@@ -5166,9 +5166,9 @@ def f(x):
         assert isinstance(x, da.Array)
         return x.sum().compute()
 
-    future = c.submit(f, x)
-    result = await future
-    assert result == sum(range(10))
+    with pytest.raises(NoCurrentClient, match=r"Future.*argument.*persist"):
+        future = c.submit(f, x)
+        result = await future
```

Comment:
If users actually provide a collection as an argument they will receive a helpful exception message. unfortunately we can only raise once the task is deserialized on the worker. That's pretty late but the best I can do right now and still much better than spurious failures as described in https://github.com/dask/distributed/issues/7498